components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1373,18 +1377,37 @@ void RewardsServiceImpl::OnPublisherBanner(std::unique_ptr<ledger::PublisherBann
     observer.OnPublisherBanner(this, new_banner);
 }
 
-void RewardsServiceImpl::OnDonate(const std::string& publisher_key, int amount, bool recurring) {
+void RewardsServiceImpl::dummy_method(ledger::Result result,
+    std::unique_ptr<ledger::PublisherInfo> info) {
+  LOG(ERROR) << "BSC]] PUBLISHER SAVED";
+}
+
+void RewardsServiceImpl::OnDonate(const std::string& publisher_key, int amount,
+  bool recurring, const ledger::PublisherInfo* publisher_info) {
   if (recurring) {
+    // If caller provided publisher info, save it to `publisher_info` table
+    if (publisher_info) {
```

Comment:
When `publisher_info` is present (optional), I'd like this to call `SavePublisherInfo`. The fields are set properly in `brave_profile_writer.cc`, but (as seen when printing below) name and URL are no longer set. I think this causes an error when trying to insert the records.

Reply:
this is fixed now